### ISSUE TITLE
Kill instance gracefully on .end()

### ISSUE THREAD

**Issue by contra** [segmentio/nightmare#409]
```
The last time you opened Electron, it unexpectedly quit while reopening windows.

Do you want to try to reopen its windows again?
```

Right now `.end()` force kills the process, and on OSX this makes the OS prompt you the next time you open it, asking if you meant to kill it and if you want to reopen the process again. I'm spinning up and closing down a ton of instance rapidly, so my OS after a while looks like this:

![](https://cldup.com/KVjMblaitc.png)

Each one of these is a lingering process that is only closed by clicking "no" - and eventually I get an ENFILE error because there are too many processes open.

**Comment by contra** [comment]
I think killing with a SIGHUP will fix this.

A temporary workaround:

```
defaults write -app "Electron" NSQuitAlwaysKeepsWindows -bool false
defaults write com.apple.CrashReporter DialogType none
```

**Comment by matthewmueller** [comment]
haha woah. really surprised that this hasn't come up before. needs more investigation

**Comment by rosshinkley** [comment]
I believe this was fixed in #487.  @contra can you verify?

**Comment by contra** [comment]
Can't confirm, not using it rn - will close though and if anybody else hits this we can reopen - thanks.
